.github/workflows/sonarqube.yml

```
@@ -0,0 +1,83 @@
+
+      - name: Compile FFmpeg and BMF if not cached
+        run: |
+          if [ ! -f "/usr/local/bin/ffmpeg" ]; then
+            echo "FFmpeg not found, starting build..."
+            (cd bmf && git checkout fork_by_oc && sudo scripts/build_ffmpeg.sh nasm yasm x264 x265 opus && ./build.sh )
+          else
+            echo "FFmpeg is already installed, skipping build."
+          fi
+
+      - name: Set up BMF
+        run: |
```

actionlint (1.7.4)

shellcheck reported issue in this script: SC2129:style:2:1: Consider using { cmd1; cmd2; } >> file instead of individual redirects

(shellcheck)